### PR TITLE
feat: add kademlia peer reachability status

### PR DESCRIPTION
**Pull request by mrekucci**
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/2563)

### CODE REVIEW

**Review by notanatol — APPROVED** [review]

**Review by aloknerurkar — APPROVED** [review]

File: pkg/topology/kademlia/kademlia.go [hunk 1]
```diff
@@ -1182,12 +1182,22 @@ func (k *Kad) EachNeighborRev(f topology.EachPeerFunc) error {
...
 func (k *Kad) EachPeerRev(f topology.EachPeerFunc) error {
-	return k.connectedPeers.EachBinRev(f)
+	return k.connectedPeers.EachBinRev(k.onlyReachable(f))
+}
+
+// onlyReachable wraps the iterator func to filer-out unreachable peers.
+func (k *Kad) onlyReachable(f topology.EachPeerFunc) topology.EachPeerFunc {
+	return func(addr swarm.Address, po uint8) (bool, bool, error) {
+		if !k.collector.Inspect(addr).IsReachable {
+			return true, false, nil
```
Comment (istae) [inline pkg/topology/kademlia/kademlia.go:1197]: the first true stops the whole iterator, I think you just need to return false, false, nil

**Review by istae — CHANGES_REQUESTED** [review]

File: pkg/topology/kademlia/kademlia.go (hunk 1 above)
Comment (acud) [reply pkg/topology/kademlia/kademlia.go:1197]: correct :+1:

**Review by acud — APPROVED** [review]
see comment from @istae otherwise LGTM

File: pkg/topology/kademlia/kademlia.go (hunk 1 above)
Comment (mrekucci) [reply pkg/topology/kademlia/kademlia.go]: Good catch, fixed!

**Review by istae — APPROVED** [review]

**Comment by mrekucci** [comment]
Draft PR, the code was moved to #2564

**Pull request closed without merging**